cards/portfolio/addon/components/isolated.js

```diff
@@ -52,6 +62,71 @@ export default LiveIsolatedCard.extend({
     return currencyGroups;
   }),
 
+  init() {
+    this._super(arguments);
+
+    if (this.web3.isMetamask) {
+      this.getMetamaskWallet.perform();
+    }
+  },
+
+  getMetamaskWallet: task(function * () {
+    let accounts = yield this.web3.eth.getAccounts();
+    if (accounts.length) {
+      let metamaskWallet = yield this.store.findRecord('wallet', 'metamask-wallet');
+
+      this.set('metamaskWallet', metamaskWallet);
+
+
+      accounts.forEach(account => {
+        this.getAssetsForWallet.perform(account);
```

Comment:
you should yield this, otherwise you'll leak async (a `for (let account of accounts) {` works nice here for that)